src/sort.rs

```diff
@@ -2,6 +2,10 @@ use std::{cmp::Ordering, collections::BTreeMap, iter::FromIterator};
 
 use toml_edit::{Array, Decor, DocumentMut, Item, RawString, Table, Value};
 
+// Leading string for combining keys such as
+// `[target.'cfg(target_os="linux")'.dependencies]` in Cargo.toml files.
+const TARGET: &str = "target";
```

Comment:
Isn't the `MATCHER` we have below conceptually already something similar? If I understand TOML correctly, then `target.cfg.dependencies` are technically nested tables right? So it is conceptually the same thing as `workspace.dependencies` except that we have a dynamic heading inbetween. Can we merge these two concepts by extending `Matcher`?

Reply:
Even if "target" is put into MATCHER, when traversing MATCHER, the situation of "target" still needs to be judged and special processing needs to be performed, so there is no change.